### PR TITLE
Fix for ASB v2's auditEnsureDotDoesNotAppearInRootsPath

### PR DESCRIPTION
**Pull request by MariusNi**
## Description

Fix for ASB v2's auditEnsureDotDoesNotAppearInRootsPath to skip PATH instances on commented out lines

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/common/commonutils/CommonUtils.h [hunk 1]
```diff
@@ -96,7 +96,7 @@ int ReplaceMarkedLinesInFile(const char* fileName, const char* marker, const cha
 int FindTextInFile(const char* fileName, const char* text, void* log);
 int CheckTextIsFoundInFile(const char* fileName, const char* text, char** reason, void* log);
 int CheckTextIsNotFoundInFile(const char* fileName, const char* text, char** reason, void* log);
-int CheckMarkedTextNotFoundInFile(const char* fileName, const char* text, const char* marker, char** reason, void* log);
+int CheckMarkedTextNotFoundInFile(const char* fileName, const char* text, const char* marker, char commentCharacter, char** reason, void* log);
```
Comment (AhmedBM) [inline src/common/commonutils/CommonUtils.h:99]: Can there only be one comment character? What if there are multiple characters?

File: src/common/commonutils/FileUtils.c [hunk 2]
```diff
@@ -1160,23 +1155,25 @@ int CheckMarkedTextNotFoundInFile(const char* fileName, const char* text, const
                 }
                 else if (0 == isalpha(found[0]))
                 {
-                    OsConfigLogInfo(log, "CheckMarkedTextNotFoundInFile: '%s' containing '%s' found in '%s' ('%s')", text, marker, fileName, found);
-                    OsConfigCaptureReason(reason, "'%s' containing '%s' found in '%s' ('%s')", text, marker, fileName, found);
+                    OsConfigLogInfo(log, "CheckMarkedTextNotFoundInFile: '%s' containing '%s' found in '%s' uncommented with '%c'", text, marker, fileName, commentCharacter);
```
Comment (AhmedBM) [inline src/common/commonutils/FileUtils.c:1158]: shouldn't it say `... uncomment by removing '...'` and not `... uncomment with '...'`

Comment (MariusNi) [reply src/common/commonutils/FileUtils.c:1158]: It's 'uncommented'.  Blah found in file foo, Uncommented with '#', for example.

Comment (AhmedBM) [reply src/common/commonutils/FileUtils.c:1158]: i see, thanks for clarifying, makes sense

**Review by AhmedBM — APPROVED** [review]
👌

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**